model_server_tools/model_packaging/export_model_utils.py

```diff
@@ -0,0 +1,289 @@
+
+MODEL_ARCHIVE_EXTENSION = '.mar'
+MODEL_SERVER_VERSION = '1.0'
+MODEL_ARCHIVE_VERSION = model_packaging.__version__
+MANIFEST_FILE_NAME = 'MANIFEST.json'
+MAR_INF = 'MAR-INF'
+ONNX_TYPE = '.onnx'
+
+
+class ModelExportUtils(object):
+    """
+    Helper utils for Model Export tool.
+    This class lists out all the methods such as validations for model archiving, ONNX model checking etc.
+    This is to keep the code in export_model.py clean and simple.
+    """
+
+    @staticmethod
+    def check_mar_already_exists(model_name, export_file, overwrite):
+        """
+        Function to check if .mar already exists
+        :param model_name:
+        :param export_file:
+        :param model_archive_extension:
+        :return:
+        """
+        if export_file is None:
+            export_file = os.path.join(os.getcwd(), '{}{}'.format(model_name, MODEL_ARCHIVE_EXTENSION))
+
+        if os.path.exists(export_file):
+            if overwrite:
+                # pylint: disable=deprecated-method
+                logging.warn("%s already exists. It will be overwritten since --force/-f was specified", export_file)
+
+            else:
+                logging.error("%s already exists. Since no --force/-f was specified, it will not be overwritten. "
+                              "Exiting the program here. Specify --force/-f flag to overwrite the %s file. "
+                              "See -h/--help for more details", export_file, export_file)
+
+                sys.exit(1)
+
+        return export_file
+
+    @staticmethod
+    def check_custom_model_types(model_path):
+        """
+        This functions checks whether any special handling is required for custom model extensions such as
+        .onnx, or in the future, for Tensorflow and PyTorch extensions.
+        :param model_path:
+        :return:
+        """
+        temp_files = [] # List of temp files added to handle custom models
+        files_to_exclude = [] # List of files to be excluded from .mar packaging.
+
+        files_set = set(os.listdir(model_path))
+        onnx_file = ModelExportUtils.find_unique(files_set, ONNX_TYPE)
+        if onnx_file is not None:
+            symbol_file, params_file = ModelExportUtils.convert_onnx_model(model_path, onnx_file)
+            files_to_exclude.append(onnx_file)
+            temp_files.append(os.path.join(model_path, symbol_file))
+            temp_files.append(os.path.join(model_path, params_file))
+
+        # More cases will go here as an if-else block
+
+        return temp_files, files_to_exclude
+
+    @staticmethod
+    def find_unique(files, suffix):
+        """
+        Function to find unique model params file
+        :param files:
+        :param suffix:
+        :return:
+        """
+        match = [f for f in files if f.endswith(suffix)]
+        count = len(match)
+
+        if count == 0:
+            return None
+        elif count == 1:
+            return match[0]
+        else:
+            params = {
```

Comment:
We only need search for .onnx file

Reply:
Okay. Will change it.

Reply:
Done in a46fa5d65e8493d56fda684f0cca7f0e97d7cf5c